models/repo.go

```diff
@@ -1408,6 +1408,13 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
 			}
 		}
 
+		// If repo has become private, we need to set its actions to private.
+		if repo.IsPrivate {
+			e.Where("repo_id = ?", repo.ID).Cols("is_private").Update(&Action{
+				IsPrivate: true,
+			})
+		}
```

Comment:
I've thought about that, and I am not sure whether having a feed that adds new events "in the past" would make much sense.